### PR TITLE
api: fix the potential race problem

### LINKED ISSUE #4872: There might be a data race problem between API v2 and dashboard
## Bug Report
Both dashboard and API v2 will set the gin mode and read it when configuring the router. But the mode here is a global variable, so we need to use `once` to prevent it from being set twice.

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

Issue Number: Close #4872.

### What is changed and how does it work?

This PR uses `once` to prevent `mode` from being set twice.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: server/apiv2/router.go [hunk 1]
```diff
@@ -34,7 +38,12 @@ const apiV2Prefix = "/pd/api/v2/"
 
 // NewV2Handler creates a HTTP handler for API.
 func NewV2Handler(_ context.Context, svr *server.Server) (http.Handler, server.ServiceGroup, error) {
-	gin.SetMode(gin.ReleaseMode)
+	once.Do(func() {
+		// These global modification will be effective only for the first invoke.
+		_ = godotenv.Load()
```
Comment (JmPotato) [inline server/apiv2/router.go:43]: What's this for?

Comment (rleungx) [reply server/apiv2/router.go:43]: Copied from the dashboard, I guess because it supports configuring the mode through environment variables.

**Review by AndreMouche — COMMENTED** [review]
LGTM

**Review by JmPotato — APPROVED** [review]

**Pull request merged**